src/platforms/linux/linuxcontroller.cpp

```diff
@@ -82,9 +82,8 @@ void LinuxController::activate(
 void LinuxController::deactivate(Reason reason) {
   logger.log() << "LinuxController deactivated";
 
-  if (reason != ReasonNone) {
-    logger.log()
-        << "No disconnect for quick server switching or connection checks";
+  if (reason == ReasonSwitching) {
+    logger.log() << "No disconnect for quick server switching";
```

Comment:
here we are in the 'controller' world: the UI component. The controller talks with the daemon. The controller is relatively stupid. For each activation/server-switching it calls deactivate + activate.  The `Reason` enum tells why we are doing this operation.

For server-switching we don't need to do the whole `deactivate` because the linux daemon is able to do server-switching (this PR). So, no deactivate() is needed for linux.